bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,660 @@
+
+                            switch (value) {
+                                case HBSTATE_WORKING:
+                                case HBSTATE_BACKMOVING:
+                                case HBSTATE_BACKMOVING_INIT:
+                                    currentCleanState = OnOffType.ON;
+                                    currentStartState = OnOffType.ON;
+                                    currentHomeState = OnOffType.OFF;
+                                    break;
+                                case HBSTATE_HOMING:
+                                case HBSTATE_DOCKING:
+                                    currentCleanState = OnOffType.OFF;
+                                    currentStartState = OnOffType.OFF;
+                                    currentHomeState = OnOffType.ON;
+                                    break;
+                                default:
+                                    currentCleanState = OnOffType.OFF;
+                                    currentStartState = OnOffType.OFF;
+                                    currentHomeState = OnOffType.OFF;
+                                    break;
+                            }
+                            updateState(CHANNEL_CLEAN, currentCleanState);
+                            updateState(CHANNEL_START, currentStartState);
+                            updateState(CHANNEL_HOME, currentHomeState);
+                        }
+                        break;
+                    case "JSON_BATTPERC":
+                        try {
+                            DecimalType battery = DecimalType.valueOf(value);
+                            if (!battery.equals(currentBattery)) {
+                                currentBattery = battery;
+                                updateState(CHANNEL_BATTERY, battery);
+                            }
+                        } catch (NumberFormatException e) {
+                            logger.debug("Couldn't parse Battery Percent.");
+                            parsingOk = false;
+                        }
+                        break;
+                    case "CPU_IDLE":
+                        if (isLinked(CHANNEL_CPU_LOAD)) {
+                            try {
+                                DecimalType cpuLoad = new DecimalType(100 - Double.valueOf(value).longValue());
+                                if (!cpuLoad.equals(currentCPULoad)) {
+                                    currentCPULoad = cpuLoad;
+                                    updateState(CHANNEL_CPU_LOAD, cpuLoad);
+                                }
+                            } catch (NumberFormatException e) {
+                                logger.debug("Couldn't parse CPU Idle.");
+                                parsingOk = false;
+                            }
+                        }
+                        break;
+                    case "LGSRV_MEMUSAGE":
+                        if (!value.equals(currentSrvMem)) {
```

Comment:
I see that you kept those checks in, I did not see a response from you. You could consider leaving them out. However, the code looks already way more readable now you use the other `updateState` method.

Reply:
I didn't want to update all states all the time if they didn't change, that was one of the reasons for keeping local states.

Reply:
> I didn't want to update all states all the time if they didn't change

I think that the framework should filter it away and it does that if I recall correctly. S

ending that same value again also has some value because it means it's still up-to-date. Combined with something like the expire-binding (an openHAB 1 concept) or [the upcoming expire profile](https://github.com/openhab/openhab-core/pull/744) which is currently discussed its crucial information that the old state is still valid.